cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -110,7 +109,11 @@ def remove_deleted_items(cls, searcher, structure_key, exclude_items):
             field_dictionary=cls._get_location_info(structure_key),
             exclude_dictionary={"id": list(exclude_items)}
         )
-        result_ids = [result["data"]["id"] for result in response["results"]]
+        if hasattr(searcher, "backend_name") and searcher.backend_name == 'meilisearch':
+            result_ids = [result["id"] for result in response["results"]]
+        else:
+            result_ids = [result["data"]["id"] for result in response["results"]]
```

Comment:
Shouldn't the `SearchEngine` subclass in `edx-search` be transforming these results so that they're the same? I'd like to find a way to get rid of this `if` condition.